cutil/parser/fsfileresource/localfs.go

```diff
@@ -0,0 +1,64 @@
+// Copyright © 2017 The Kubicorn Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fsfileresource
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"regexp"
+	"strings"
+)
+
+// LocalResource could hold explicit information local file systems
+// right now just helps us for type enforcment
+type LocalResource struct{}
+
+//NewFileLocalResource creates a new file resource fetcher
+func NewFileLocalResource(fr LocalResource) *LocalResource {
+	return &fr
+}
+
+// Read reads file from a local path and returns as []byte
+func (fr *LocalResource) Read(sourcePath string) ([]byte, error) {
+
+	// Here we check if the path starts with ~ or /
+	match, err := regexp.MatchString("[/|~]", sourcePath[:1])
+	if err != nil {
+		return []byte(""), err
+	}
+
+	if !match {
+		return []byte(""), fmt.Errorf("malformed sourcePath path expected first character to be ~ or /, but got %s", sourcePath)
+	}
+
+	// If absolutePath starts with ~ we search for $HOME
+	// and preppend it to the absolutePath overwriting the first character
+	// TODO: Add Windows support
+	if strings.HasPrefix(sourcePath, "~") {
```

Comment:
@prateekgogia "filepath.HasPrefix exists for historical compatibility and should not be used."
```go
// HasPrefix exists for historical compatibility and should not be used.
func HasPrefix(p, prefix string) bool {
    return strings.HasPrefix(p, prefix)
}
```